### ISSUE TITLE
Server is returning 451, but the email is still appearing in the inbox

### ISSUE THREAD

**Issue by elliots** [axllent/mailpit#283]
Hi, I'm sometimes getting `Message failed: 451 4.3.5 Unable to process mail`, but the message still appears. So when I retry and it works, I have two mails.

looks like maybe sending two messages at the same time, and theyre given the same tag?

```
time="2024/04/24 11:20:07" level=debug msg="[tags] adding tag \"si-on-app-requote_regression-1713921558073-00\" to BaTkmBUyYDuCmXbPZhiy8d"
time="2024/04/24 11:20:07" level=debug msg="[tags] adding tag \"si-on-app-requote_regression-1713921558073-00\" to 8g6ZAXhw6fRjRrWtwdSPCd"
time="2024/04/24 11:20:07" level=error msg="[db] error storing message: constraint failed: UNIQUE constraint failed: tags.Name (2067)"```

**Comment by axllent** [comment]
Thanks for the details @elliots. It's hard to imagine that this could happen _unless_ two separate parallel processes send an email at *exactly* the same time, both adding the same tag. Does this sound like what your end is doing?

**Comment by axllent** [comment]
I have just released v1.17.1 which has a fix for this (or at least for what I presume is causing the error). Please confirm this fixes it for you?

**Comment by elliots** [comment]
I’ll update and try it out, thankyou. I didn’t think I was setting that tag (I thought it was some built in tag set when receiving a message), but we must be! I’ll look closer.
